components/feature/qr/src/main/java/mozilla/components/feature/qr/QrFragment.kt

```diff
@@ -271,6 +280,16 @@ class QrFragment : Fragment() {
         }
     }
 
+    internal fun startExecutorService() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            backgroundExecutor = Executors.newSingleThreadExecutor()
+        }
+    }
+
+    internal fun stopExecutorService() {
+        backgroundExecutor?.shutdownNow()
```

Comment:
nit: We should nullify `backgroundExecutor = null` when we don't need it any more